### DISCUSSION TITLE
Performance issue with simple query

### DISCUSSION THREAD

**Discussion by magnusottosson** [prisma/prisma#19399] [Q&A] [answered]
Hey,

I have a many to many releation between User and Car. I just want to fetch the relation and I do that with a query like:

```
await this.prisma.carsOnUsers.findMany({
      where: { userId },
      select: { carId: true },
    })
```

Should be fast right?

I have turned on logging with:

```
this.$on('query', (e) => {
        const { query, params, duration } = e;

        this.logger.log({
          query,
          params,
          duration,
        });
      });
```
 
 this outputs the actual query and then I also have a middleware like 
 
```
 this.$use(async (params, next) => {
      const before = Date.now();

      const result = await next(params);

      const after = Date.now();

      const message = `${params.model}.${params.action}`;

      if (process.env.LOG_SQL_QUERIES === 'true') {
        this.logger.log({
          msg: message,
          duration: after - before,
        });
      }

      return result;
    });
```

The actual query takes something like 5ms on my local dev machine, but the measurement in the middleware says something like 150ms.

This is when I run a lot of stuff in parallel in a quite big GQL query. If I just select a smaller portion of the GQL stuff I can get the time on the middleware log down to around 50ms.

Is there some kind of performance issue with this query in prisma? Or is the middleware just waiting on the next node tick and I'm doing to many things at the same time?

**Comment by nurul3101** [comment, marked as answer]
Hello @magnusottosson 👋 

Are you using the latest version of Prisma i.e. `4.14.1`? If not, I would recommend you try it on the latest version.

The difference in timings between the actual query run on the database and the timings from middleware can be because the middleware code itself has some overhead, including the cost of computing timestamps, invoking the logging function, and awaiting the next operation. Also, if you're running a lot of operations in parallel, then the Node.js event loop could be busy processing other tasks between the start and end of your database query. This could lead to a delay between when the database query finishes and when your middleware code runs to compute the 'after' timestamp.

**Reply by magnusottosson** [reply]
Yes, I'm on the latest version.

Its probably just node that is busy as you say. But it is a bit strange because I have quite a few other queries and they all run in around 15ms while this one and one more takes 125ms...

Must be some kind of race condition or something somewhere. I need to dig deeper. Just wanted to check so I wasnt missing something obvious.

Thanks.

**Reply by nurul3101** [reply]
No worries! 🙂 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved.
